backend/src/main/java/com/bakdata/conquery/resources/admin/rest/AdminTablesResource.java

```diff
@@ -83,24 +88,52 @@ public Response remove(@QueryParam("force") @DefaultValue("false") boolean force
 	@Produces(AdditionalMediaTypes.JSON)
 	public List<ImportId> listImports() {
 		return namespace.getStorage()
-						.getAllImports()
-						.stream()
-						.filter(imp -> imp.getTable().equals(table))
-						.map(Import::getId)
-						.collect(Collectors.toList());
+				.getAllImports()
+				.stream()
+				.filter(imp -> imp.getTable().equals(table))
+				.map(Import::getId)
+				.collect(Collectors.toList());
 	}
 
 	@DELETE
-	@Path("imports/{"+IMPORT_ID+"}")
+	@Path("imports/{" + IMPORT_ID + "}")
 	public void deleteImport(@PathParam(IMPORT_ID) Import imp) {
 		processor.deleteImport(imp);
 	}
 
 
 	@GET
-	@Path("imports/{"+IMPORT_ID+"}")
+	@Path("imports/{" + IMPORT_ID + "}")
 	public Import getImport(@PathParam(IMPORT_ID) Import imp) {
 		return imp;
 	}
 
+	@PUT
+	@Path("imports")
+	public void updateImport(@NotNull @QueryParam(IMPORT_ID) Import imp, @NotNull @QueryParam("file") File importFile) throws IOException {
```

Comment:
Kannst du hier noch eine Endpunkt erstellen der die CQPP durch einen InputStream aufnimmt anstatt aus einem lokalen Ordner

Reply:
Und IMPORT_ID sollte hier nicht ein QueryParam sondern ein PathParam sein so wie in DELETE

Reply:
Ich weiß, es ist nicht 100% REST-Ful, aber die ImportId beim reinladen zu wissen ist relativ nervig, mir (@thoniTUB @MaxVonKnobloch  wie seht ihr das?) wäre es lieber, wenn die id direkt aus der CQPP abgeleitet wird, wie beim normalen Import.

Reply:
@thoniTUB @awildturtok  jetzt brauchen die Endpunkte für Import-Update entweder Pfad zum lokalen File oder die Datei selbst als Inputstream. Der Import selbst wird nicht mehr benötigt.

Reply:
Ich finde es für den Moment okay, aber wurde später versuchen die Tags aus dem Preprocess heraus zuhalten und dann das PUT wieder RESTful zu machen.

Reply:
Das POST naturlich auch, das sollte auf "/table/{tableid}" gehen und damit auch aus dem Import verschwinden